src/hotspot/share/nmt/mallocTracker.cpp

```diff
@@ -239,13 +239,15 @@ bool MallocTracker::print_pointer_information(const void* p, outputStream* st) {
   const MallocHeader* likely_live_block = nullptr;
   {
     const size_t smallest_possible_alignment = sizeof(void*);
-    const uint8_t* here = align_down(addr, smallest_possible_alignment);
-    const uint8_t* const end = here - (0x1000 + sizeof(MallocHeader)); // stop searching after 4k
+    uintptr_t here = (uintptr_t)align_down(addr, smallest_possible_alignment);
+    if (here == 0) {
```

Comment:
We should use `nullptr` here not `0`:

if (here == nullptr) {

Reply:
but `here` is of `uintptr_t` datatype.